components/common/IdentityForm.vue

```diff
@@ -147,23 +179,29 @@ const enhanceIdentityData = (): Record<string, any> => {
 }
 
 const fetchDeposit = async () => {
-  const api = await apiInstance.value
+  const api = await identityApi.value
   return api.consts.identity?.basicDeposit?.toString()
 }
 
-const submit = async (): Promise<void> => {
-  const api = await apiInstance.value
+const submit = async (isDelete?: boolean): Promise<void> => {
+  const api = await identityApi.value
   initTransactionLoader()
-  const cb = api.tx.identity.setIdentity
-  const args = [enhanceIdentityData()]
-  howAboutToExecute(accountId.value, cb, args, onSuccess)
-}
-
-const onSuccess = (block: string) => {
-  showNotification(
-    `[Identity] You are known as ${identity.value.display} since block ${block}`,
-    notificationTypes.success
-  )
+  const cb = isDelete
+    ? api.tx.identity.clearIdentity
+    : api.tx.identity.setIdentity
+  const args = isDelete ? [] : [enhanceIdentityData()]
+  howAboutToExecute(accountId.value, cb, args, (block: string) => {
+    if (isDelete) {
+      identity.value = {}
+      refetchIdentity()
+    }
+    showNotification(
+      isDelete
+        ? `[Identity] You have cleared your account's identity since block ${block}`
+        : `[Identity] You are known as ${identity.value.display} since block ${block}`,
+      notificationTypes.success
+    )
+  })
 }
```

Comment:
I would make it as a two separated functions

Reply:
done